src/Extensions/Cassandra.AppMetrics/Implementations/HdrHistogramReservoir.cs

```diff
@@ -0,0 +1,182 @@
+﻿//
+//      Copyright (C) DataStax Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+using App.Metrics.Concurrency;
+using App.Metrics.ReservoirSampling;
+using Cassandra.AppMetrics.HdrHistogram;
+
+namespace Cassandra.AppMetrics.Implementations
+{
+    /// <summary>
+    /// Adapted code from
+    /// https://github.com/alhardy/AppMetrics.Reservoirs/tree/3a1861386ac73b21b37e548c61575602eed67f89/src/App.Metrics.Extensions.Reservoirs.HdrHistogram
```

Comment:
I think we must include the original license header.

We can also include a NOTICE file (similar to the one on the core driver) with the same information and linking to the license.

Reply:
Since it's the same repo I added two more "entries" to the already existing NOTICE file.